src/auto/functions.rs

```diff
@@ -24,46 +28,120 @@ use Resource;
-//}))
-//}
+pub fn bus_get_future(
+    bus_type: BusType,
+) -> Pin<Box_<dyn std::future::Future<Output = Result<DBusConnection, glib::Error>> + 'static>> {
+    Box_::pin(crate::GioFuture::new(&(), move |_obj, send| {
+        let cancellable = Cancellable::new();
+        bus_get(bus_type, Some(&cancellable), move |res| {
+            send.resolve(res);
+        });
 
-//pub fn bus_get_sync<P: IsA<Cancellable>>(bus_type: /*Ignored*/BusType, cancellable: Option<&P>) -> Result</*Ignored*/DBusConnection, glib::Error> {
-//    unsafe { TODO: call gio_sys:g_bus_get_sync() }
-//}
+        cancellable
+    }))
+}
 
-//pub fn bus_own_name(bus_type: /*Ignored*/BusType, name: &str, flags: /*Ignored*/BusNameOwnerFlags, bus_acquired_handler: /*Unimplemented*/Fn(/*Ignored*/DBusConnection, &str), name_acquired_handler: /*Unimplemented*/Fn(/*Ignored*/DBusConnection, &str), name_lost_handler: /*Unimplemented*/Fn(/*Ignored*/DBusConnection, &str), user_data: /*Unimplemented*/Option<Fundamental: Pointer>) -> u32 {
+pub fn bus_get_sync<P: IsA<Cancellable>>(
+    bus_type: BusType,
+    cancellable: Option<&P>,
+) -> Result<DBusConnection, glib::Error> {
+    unsafe {
+        let mut error = ptr::null_mut();
+        let ret = gio_sys::g_bus_get_sync(
+            bus_type.to_glib(),
+            cancellable.map(|p| p.as_ref()).to_glib_none().0,
+            &mut error,
+        );
+        if error.is_null() {
+            Ok(from_glib_full(ret))
+        } else {
+            Err(from_glib_full(error))
+        }
+    }
+}
+
+//pub fn bus_own_name(bus_type: BusType, name: &str, flags: BusNameOwnerFlags, bus_acquired_handler: Option<Box_<dyn Fn(&DBusConnection, &str) + 'static>>, name_acquired_handler: Option<Box_<dyn Fn(&DBusConnection, &str) + 'static>>, name_lost_handler: Option<Box_<dyn Fn(&DBusConnection, &str) + 'static>>) -> u32 {
 //    unsafe { TODO: call gio_sys:g_bus_own_name() }
 //}
 
-//pub fn bus_own_name_on_connection(connection: /*Ignored*/&DBusConnection, name: &str, flags: /*Ignored*/BusNameOwnerFlags, name_acquired_handler: /*Unimplemented*/Fn(/*Ignored*/DBusConnection, &str), name_lost_handler: /*Unimplemented*/Fn(/*Ignored*/DBusConnection, &str), user_data: /*Unimplemented*/Option<Fundamental: Pointer>) -> u32 {
+//pub fn bus_own_name_on_connection(connection: &DBusConnection, name: &str, flags: BusNameOwnerFlags, name_acquired_handler: Option<Box_<dyn Fn(&DBusConnection, &str) + 'static>>, name_lost_handler: Option<Box_<dyn Fn(&DBusConnection, &str) + 'static>>) -> u32 {
 //    unsafe { TODO: call gio_sys:g_bus_own_name_on_connection() }
 //}
 
-//pub fn bus_own_name_on_connection_with_closures(connection: /*Ignored*/&DBusConnection, name: &str, flags: /*Ignored*/BusNameOwnerFlags, name_acquired_closure: /*Ignored*/Option<&glib::Closure>, name_lost_closure: /*Ignored*/Option<&glib::Closure>) -> u32 {
-//    unsafe { TODO: call gio_sys:g_bus_own_name_on_connection_with_closures() }
-//}
+pub fn bus_own_name_on_connection_with_closures(
```

Comment:
This should be a) renamed to not include the "with_closures" part and b) the closures have fixed signatures so it would be better for users of this API to instead take proper Rust closures with the correct types so users don't have to work with `glib::Value`s :)